src/CodeExampleLanguageProcessing.ts

```diff
@@ -0,0 +1,371 @@
+  },
+  {
+    directiveValue: "php",
+    canonicalValue: CanonicalLanguageValues.PHP,
+    extension: LanguageFileExtensions.PHP,
+  },
+  {
+    directiveValue: "python",
+    canonicalValue: CanonicalLanguageValues.PYTHON,
+    extension: LanguageFileExtensions.PYTHON,
+  },
+  {
+    directiveValue: "ruby",
+    canonicalValue: CanonicalLanguageValues.RUBY,
+    extension: LanguageFileExtensions.RUBY,
+  },
+  {
+    directiveValue: "rust",
+    canonicalValue: CanonicalLanguageValues.RUST,
+    extension: LanguageFileExtensions.RUST,
+  },
+  {
+    directiveValue: "scala",
+    canonicalValue: CanonicalLanguageValues.SCALA,
+    extension: LanguageFileExtensions.SCALA,
+  },
+  {
+    directiveValue: "sh",
+    canonicalValue: CanonicalLanguageValues.SHELL,
+    extension: LanguageFileExtensions.SHELL,
+  },
+  {
+    directiveValue: "shell",
+    canonicalValue: CanonicalLanguageValues.SHELL,
+    extension: LanguageFileExtensions.SHELL,
+  },
+  {
+    directiveValue: "swift",
+    canonicalValue: CanonicalLanguageValues.SWIFT,
+    extension: LanguageFileExtensions.SWIFT,
+  },
+  {
+    directiveValue: "text",
+    canonicalValue: CanonicalLanguageValues.TEXT,
+    extension: LanguageFileExtensions.TEXT,
+  },
+  {
+    directiveValue: "typescript",
+    canonicalValue: CanonicalLanguageValues.TYPESCRIPT,
+    extension: LanguageFileExtensions.TYPESCRIPT,
+  },
+  {
+    directiveValue: "undefined",
+    canonicalValue: CanonicalLanguageValues.UNDEFINED,
+    extension: LanguageFileExtensions.UNDEFINED,
+  },
+  {
+    directiveValue: "xml",
+    canonicalValue: CanonicalLanguageValues.XML,
+    extension: LanguageFileExtensions.XML,
+  },
+  {
+    directiveValue: "yaml",
+    canonicalValue: CanonicalLanguageValues.YAML,
+    extension: LanguageFileExtensions.YAML,
+  },
+  {
+    directiveValue: "json\\n :copyable: false",
+    canonicalValue: CanonicalLanguageValues.JSON,
+    extension: ".json",
+  },
+  {
+    directiveValue: "json\\n:copyable: true",
+    canonicalValue: CanonicalLanguageValues.JSON,
+    extension: ".json",
+  },
+];
+
+export const getLanguageDetailsForCodeNode = (
+  node: CodeNode
+): LanguageMapper => {
+  if (node.lang != null) {
+    const mapping = languageValueMappings.find(
+      (mapper) => mapper.directiveValue === node.lang
+    );
+    if (mapping != undefined) {
+      return mapping;
+    } else {
+      return {
+        directiveValue: "none",
+        canonicalValue: CanonicalLanguageValues.UNDEFINED,
+        extension: LanguageFileExtensions.UNDEFINED,
+      };
+    }
+  } else {
+    return {
+      directiveValue: "none",
+      canonicalValue: CanonicalLanguageValues.UNDEFINED,
+      extension: LanguageFileExtensions.UNDEFINED,
+    };
+  }
+};
+
+export const getLanguageDetailsForLiteralInclude = (
+  node: DirectiveNode
+): LanguageMapper => {
+  const maybeChildNode = node.children[0] as CodeNode;
+  const maybeNodeChildLang = maybeChildNode.lang;
```

Comment:
Let's not assume children[0] exists
```suggestion
  const maybeNodeChildLang = maybeChildNode?.lang;
```